### ISSUE TITLE
"Phone: Optional/Required" setting in Customizer (from core WC) should indicate if it's being overridden by PPEC "Require Phone Number" Setting

### ISSUE THREAD

**Issue by dphyled** [woocommerce/woocommerce-gateway-paypal-express-checkout#516]
"Require Phone Number" setting in PayPal Checkout overrides the WooCommerce core setting in Customizer under **WooCommerce > Checkout**, however there is no way to tell when you are in Customizer. This can lead to confusion.

For example in the following situation: 
1. You have PayPal Checkout active, and have not enabled the "Require Phone Number" setting (default):
![Screenshot](https://d.pr/YmevEd+)
Screenshot direct link: https://d.pr/YmevEd
2. You want phone to be required, so you go to Customizer > WooCommerce > Checkout, and set the phone to required:
![Screenshot](https://d.pr/NMhgpK+)
Screenshot direct link: https://d.pr/NMhgpK
3. You do some testing, and the phone is still not required, but you are unsure why.

Adding a little note under the customizer field when PPEC is active would avoid this confusion.

**Comment by TimBHowe** [comment]
Ran into the same issue on a client site. We have created a simple pull request to disable and override the "Require Phone Number" setting in the plugin in favor of the Customizer settings if it is set to "Hidden" or "Required".

Hope this helps.

**Comment by TimBHowe** [comment]
I have created a new PR.

The Customizer allow you to set the phone field as "Hidden", "Optional" or "Required".

The PPEC plugin only allows it to be "Optional" or "Required".

In the case where it is "Required" on the WC Customizer if we let the PPEC plugin setting override it the user would need to set the PPEC plugin as "Required" as well to actually require the phone number. This can lead to confusion and make the WC Customizer unnecessary or the PPEC plugin setting to be redundant.

In the case where it is "Hidden" on the WC Customizer and "Required" in the PPEC plugin the PPEC plugin tries to set the nonexistent billing_phone field as required in the includes/class-wc-gateway-ppec-checkout-handler.php#L171 and when it notices it is not there it skips it. In this case it could be very confusing to users to allow then to override the Customizer settings with the PPEC settings.

The PPEC plugin should probably just get rid of the require phone setting and use the Customizer setting in it's place. I have created an alternative pull request to show this solution and I think it is the better option.

Remove the "Require Phone Number" setting from the woocommerce-gateway-paypal-express-checkout plugin in favor of using the setting in the Cusomizer > WooCommerce > Checkout setting.

We have included instructions and a UI directing users to set it in the Customizer.

**Comment by mattallan** [comment]
Closing this issue since we have fixed the bug related to the "Require Phone number" setting. See https://github.com/woocommerce/woocommerce-gateway-paypal-express-checkout/pull/697.

---

The "Require Phone Number" setting should've only be used during an active Paypal session which is when customers return from PayPal after using the Smart Buttons on your Product or Cart pages.

With this setting enabled, if the customer returns from PayPal without filling in their phone number, they will be prompted to fill it in: 

![image](https://user-images.githubusercontent.com/2275145/77504244-8c44bf00-6eab-11ea-9da5-fa26e60919eb.png)
